app/Models/Forum/Post.php

```diff
@@ -43,6 +47,19 @@ class Post extends Model
 
     private $normalizedUsers = [];
 
+    protected static function boot()
+    {
+        parent::boot();
+
+        if (!priv_check('ForumTopicModerate')->can()) {
```

Comment:
Last I checked it's only needed in `TopicController#show`, `PostsController#destroy`, and the post position thingy (Topics's `nthPost`, `postPosition` and `postsPosition`).

The rest (cache update, topic move, etc) seem to require explicit with/without deleted post regardless of current user.